openedx/openedx-platform
# feat: Adds a new discussion topic configuration mechanism [BD-38] [TNL-8623] [BB-4968]

## Description

The new discussion configuration system links discussion topics directly to the course structure. This change adds a new task that sends a discussion update signal if there are any changes to the course. This signal includes all the context needed to update the configuration of the course.

The handler for this new event will create a new entry for each unit that needs a topic in the database. In the future this will be used to see the topics in the course.

## Dependencies
This PR build on a few other PRs:
- https://github.com/edx/edx-platform/pull/28749
- https://github.com/edx/edx-platform/pull/28864

The relevant commit is currently: 11de31553ca4e0f1dc816ae88235e41160e1d5f3

## Supporting information

- [TNL-8623](https://openedx.atlassian.net/browse/TNL-8623)
- ADR:  https://github.com/edx/edx-platform/pull/28221

## Testing instructions

- For any existing course, change any setting and publish. 
- You should see new DiscussionTopicLink objects, one for each general topic and one for each unit. 
- If there are any graded discussions they will not be included.
- Try changing the settings and make sure that the behaviour in the ADR is matched. 
- Try importing and exporting the course. The settings should be retained. 
- Try exporting a course and importing it as a new course. The discussion config model should be created to match the original settings. 

## Deadline

"None"
